UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/FulfilmentDataService.cs

```diff
@@ -211,5 +176,104 @@ private void CreateIsoAndSha1ForExchangeSet(List<FssBatchFile> fileDetails, stri
+            List<FssBatchFile> files = new();
+
+            if (!string.IsNullOrEmpty(essBatchId))
+            {
+                FssBatchStatus fssBatchStatus = await _fssService.CheckIfBatchCommitted(essBatchId);
+
+                if (fssBatchStatus == FssBatchStatus.Committed)
+                {
+                    _fileSystemHelper.CreateDirectory(downloadPath);
+                    files = await GetBatchFiles(essBatchId);
+                    DownloadFiles(files, downloadPath);
+                }
+                else
+                {
+                    _logger.LogError(EventIds.FssPollingCutOffTimeout.ToEventId(), "Batch is not committed within given polling cut off time | {DateTime} | Batch Status : {BatchStatus} | _X-Correlation-ID : {CorrelationId}", DateTime.Now.ToUniversalTime(), fssBatchStatus, CommonHelper.CorrelationID);
+                    throw new FulfilmentException(EventIds.FssPollingCutOffTimeout.ToEventId());
+                }
+            }
+            return (downloadPath, files);
+        }
+
+        private async Task<List<FssBatchFile>> GetBatchFiles(string essBatchId)
+        {
+            List<FssBatchFile> batchFiles = null;
+
+            GetBatchResponseModel batchDetail = await _fssService.GetBatchDetails(essBatchId);
+            batchFiles = batchDetail.Files.Select(a => new FssBatchFile { FileName = a.Filename, FileLink = a.Links.Get.Href }).ToList();
+
+            if (!batchFiles.Any() || batchFiles.Any(f => f.FileName.ToLower().Contains("error")))
+            {
+                _logger.LogError(EventIds.ErrorFileFoundInBatch.ToEventId(), "Either no files found or error file found in batch with BathcID - {BatchID} | {DateTime} | _X-Correlation-ID:{CorrelationId}", essBatchId, DateTime.Now.ToUniversalTime(), CommonHelper.CorrelationID);
+                throw new FulfilmentException(EventIds.ErrorFileFoundInBatch.ToEventId());
+            }
+            return batchFiles;
+        }
+
+        private void DownloadFiles(List<FssBatchFile> fileDetails, string downloadPath)
+        {
+            Parallel.ForEach(fileDetails, file =>
+            {
+                string filePath = Path.Combine(downloadPath, file.FileName);
+                Stream stream = _fssService.DownloadFile(file.FileName, file.FileLink).Result;
+                _fileSystemHelper.CreateFileCopy(filePath, stream);
+            });
+        }
+
+        private async Task<bool> CreatePosBatch(string downloadPath, string fileExtension, string mediaType, Batch batchType)
+        {
+            bool isBatchCreated = false;
+
+            string batchId = await _fssService.CreateBatch(mediaType, batchType);
+            IEnumerable<string> filePaths = _fileSystemHelper.GetFiles(downloadPath, fileExtension, SearchOption.TopDirectoryOnly);
+            UploadBatchFiles(filePaths, batchId);
+            bool isCommitted = await _fssService.CommitBatch(batchId, filePaths);
+
+            if (isCommitted)
```

Comment:
Why not just return isCommitted and remove isBatchCreated completely?

Reply:
Resolved